cmd/kube-scheduler/app/options/options_test.go

```diff
@@ -433,6 +459,20 @@ pluginConfig:
 			options:       &Options{},
 			expectedError: "no configuration has been provided",
```

Comment:
Any chance that this and the rest of expectedError can be migrated to `checkErrFn`?

Reply:
IMO it's pretty useful to have these be separate.
We could add a comment explaining the value. (they're both convenient and optional)

Otherwise your functions could end up looking like this:
```go
expectedError: func(err error) bool {
    if ! strings.Contains("my expected string") {
        return false
    }
    return runtime.IsStrictDecodingError(err)
}
```
Note how:
- `runtime.IsStrictDecodingError` now has to be wrapped
- The function has to branch to support both cases
- `string.Contains` is not using assert
  ( If it was, you'd have to close scope on `t`, and you wouldn't return it or the assertion could be conflated with the function. You could change the signature to take in `t` and not return anything, but then you'd always have to wrap common type-check functions for errors. )

Since checking for error string contents is so common and the runner code is simple to maintain, I think it's fine to support both fields.

Reply:
I wouldn't advocate for wrapping `string.Contains`, but I was hoping the current errors have some type associated with them.